mujoco_warp/_src/passive.py

```diff
@@ -161,6 +161,92 @@ def _qfrc_passive_gravcomp(
   qfrc_passive_out[worldid, dofid] += qfrc_gravcomp_in[worldid, dofid]
 
 
+@wp.kernel
+def _flex_elasticity(
+  # Model:
+  body_dofadr: wp.array(dtype=int),
+  flex_dim: wp.array(dtype=int),
+  flex_vertadr: wp.array(dtype=int),
+  flex_edgeadr: wp.array(dtype=int),
+  flex_elemedgeadr: wp.array(dtype=int),
+  flex_vertbodyid: wp.array(dtype=int),
+  flex_elem: wp.array(dtype=int),
+  flex_elemedge: wp.array(dtype=int),
+  flexedge_length0: wp.array(dtype=float),
+  flex_stiffness: wp.array(dtype=float),
+  flex_damping: wp.array(dtype=float),
+  # Data in:
+  flexvert_xpos_in: wp.array2d(dtype=wp.vec3),
+  flexedge_length_in: wp.array2d(dtype=float),
+  flexedge_velocity_in: wp.array2d(dtype=float),
+  # In:
+  timestep: float,
+  # Data out:
+  qfrc_spring_out: wp.array2d(dtype=float),
+):
+  worldid, t = wp.tid()
+  f = 0  # TODO(quaglino): this should become a function of t
+
+  # TODO(quaglino): support dim != 2
+  dim = flex_dim[f]
+  nedge = 3
+  nvert = 3
+  kD = flex_damping[f] / timestep
+
+  edges = wp.mat(1, 2, 2, 0, 0, 1, shape=(3, 2), dtype=int)
+  gradient = wp.mat(0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, shape=(3, 6))
```

Comment:
nit: I think these inits let you do a shortcut and will fill if you pass a single scalar, so:

```py
gradient = wp.mat(0.0, shape=(3,6))
```

Reply:
Done.